Tutorial: Update canary release tutorial (#6717)
PostHog/posthog.com: contents/tutorials/canary-release.md

This tutorial explains what a canary release is, and how to set one up and monitor it in PostHog.

## What is a canary release?

A canary release or canary deployment is the process of rolling out a new feature to a subset of users before releasing it to a larger group. Developers check the new feature is working without issues on the limited group. They watch for issues and analyze usage metrics to confirm. Once satisfied with tests and analysis, the feature rolls out to a larger group (or everyone).

![Canary release](../images/tutorials/canary-release/canary.png)

The name comes from the phrase “canary in a coal mine” which alludes to miners bringing a canary into mines with them. If there were toxic gases, the canary would die and stop chirping, giving an early warning to the miners to get out before they do the same. Testing on a small group of users acts as a canary for issues with the feature, preventing those issues from affecting a larger group.

## Prerequisites for canary releases

> To set up a canary release in PostHog, we use feature flags, but before we set them up, we need user identification.

Reviewer: ```suggestion
To set up a canary release in PostHog, we use feature flags. However, in order to do this, we need to set up user identification.
```
